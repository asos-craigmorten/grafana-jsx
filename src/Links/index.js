import PropTypes from "prop-types";

const Links = ({ children }) => ({
  links: children,
});

Links.PropTypes = {
  children: PropTypes.arrayOf(PropTypes.object),
};

export default Links;
