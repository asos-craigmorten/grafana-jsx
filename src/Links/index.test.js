import Links from "./";

describe("Links", () => {
  describe("when passed no Link children", () => {
    it("should render", () => {
      expect(<Links />).toMatchSnapshot();
    });
  });

  describe("when passed Link children", () => {
    it("should render", () => {
      expect(
        <Links>
          <test-link test-link-prop={Symbol("test-link-value")} />
          <test-link test-link-prop={Symbol("test-link-value")} />
        </Links>
      ).toMatchSnapshot();
    });
  });
});
