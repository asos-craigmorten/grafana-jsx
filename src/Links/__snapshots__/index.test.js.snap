// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links when passed Link children should render 1`] = `
Object {
  "links": Array [
    Object {
      "test-link-prop": Symbol(test-link-value),
    },
    Object {
      "test-link-prop": Symbol(test-link-value),
    },
  ],
}
`;

exports[`Links when passed no Link children should render 1`] = `
Object {
  "links": Array [],
}
`;
